scanspec/service.py

```diff
@@ -1,19 +1,114 @@
-from typing import Any
+import base64
+from dataclasses import dataclass
+from typing import Any, List, Optional
 
 import aiohttp_cors
+import graphql
 from aiohttp import web
-from apischema.graphql import graphql_schema
+from apischema.graphql import graphql_schema, resolver
 from graphql_server.aiohttp.graphqlview import GraphQLView, _asyncify
+from numpy import array2string, dtype, float64, frombuffer, ndarray
 
+from scanspec.core import Path
 from scanspec.specs import Spec
 
+# See confluence page for clarity on naming conventions and data structure:
+# https://confluence.diamond.ac.uk/display/SSCC/Data+Structure+and+Naming+Conventions
 
+
+# The lowest query level
+@dataclass
+class Points:
+    def __init__(self, points: Optional[ndarray]):
+        self._points = points
+
+    @resolver
+    def string(self) -> Optional[str]:
+        return array2string(self._points)
+
+    @resolver
+    def float_list(self) -> Optional[List[float]]:
+        if self._points is None:
+            return None
+        else:
+            return self._points.tolist()
+
+    @resolver
+    def b64(self) -> Optional[str]:
+        if self._points is None:
+            return None
+        else:
+            # make sure the data is sent as float64
+            assert dtype(self._points[0]) == dtype(float64)
+            return base64.b64encode(self._points.tobytes()).decode("utf-8")
+
+    # Self b64 decoder for testing purposes
+    @resolver
+    def b64Decode(self) -> Optional[str]:
+        if self._points is None:
+            return None
+        else:
+            r = dtype(self._points[0])
+            s = base64.decodebytes(base64.b64encode(self._points.tobytes()))
+            t = frombuffer(s, dtype=r)
+            return array2string(t)
+
+
+@dataclass
+class axisFrames:
```

Comment:
```suggestion
class AxisFrames:
```

Reply:
Annoyingly there are 3 naming conventions for Python:
- `functions_and_methods_are_snake_case()`
- `class ClassNamesAreInitialCapsCamel:`
- `CONSTANTS_ARE_ALL_CAPS=1`

Reply:
Understood, made a mental (and physical) note of this

Name updated as suggested